src/main/groovy/life/qbic/utils/NanoporeParser.groovy

```diff
@@ -171,6 +177,24 @@ class NanoporeParser {
             return convertedDirectory
         }
 
+        private static Map convertToRelativePaths(Map content, Path root) {
+            content["path"] = toRelativePath(content["path"] as String, root)
+            if (content["children"]) {
+                // Children always contains a map, so convert recursively
+                content["children"] = (content["children"] as List).collect { convertToRelativePaths(it as Map, root) }
+            }
+            return content
+
+        }
+
+        private static String toRelativePath(String path, Path root) {
+            if (root.toString() == path) {
```

Comment:
String to Path comparison.
```suggestion
            if (root.toString().equals(path.toString())) {
```

Reply:
path is already a String, thus you can use the equal operator ;)

Reply:
Nah should still use `equals` at least to be sure.

Reply:
In Java, you have to be careful since the equality operator compares the objects of equality and the equals method compares the content of strings. Here we want to compare the content.